### PR TITLE
Drawing Classifier: Improved error handling for Validation Set

### LINKED ISSUE #1615: Validation set option for drawing classifier has the following issues
- When set to a string that is not 'auto', it throws an assert error instead of a simple error message

```
In [4]: model = tc.drawing_classifier.create(train_data, 'label', validation_set='foo')
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-4-903a8934d052> in <module>()
----> 1 model = tc.drawing_classifier.create(train_data, 'label', validation_set='foo')

/Users/srikris/.virtualenvs/turicreate-5.0/lib/python2.7/site-packages/turicreate/toolkits/drawing_classifier/drawing_classifier.pyc in create(input_dataset, target, feature, validation_set, pretrained_model_url, batch_size, max_iterations, verbose)
    156             validation_set, feature) if is_validation_stroke_input else validation_set
    157     elif isinstance(validation_set, str):
--> 158         assert (validation_set == 'auto')
    159         if dataset.num_rows() >= 100:
    160             if verbose:

AssertionError:

In [5]:
```

- When a dataset with < 100 rows is put in, you get a validation set of NaN. Likely due to a bug in handling. This happens during the default dataset. 

```
+------------------------+------------------------+------------------------+------------------------+------------------------+
| Iteration              | Training Loss          | Training Accuracy      | Validation Accuracy    | Elapsed Time (seconds) |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 6                      | 0.653                  | 0.973                  | nan                    | 3.992                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 13                     | 0.618                  | 0.973                  | nan                    | 4.295                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 20                     | 0.581                  | 0.973                  | nan                    | 4.602                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 27                     | 0.544                  | 0.973                  | nan                    | 4.902                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 34                     | 0.503                  | 0.973                  | nan                    | 5.198                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 41                     | 0.463                  | 0.973                  | nan                    | 5.497                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 48                     | 0.429                  | 0.973                  | nan                    | 5.803                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 55                     | 0.399                  | 0.973                  | nan                    | 6.105                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 62                     | 0.382                  | 0.973                  | nan                    | 6.407                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 69                     | 0.368                  | 0.973                  | nan                    | 6.709                  |
+------------------------+------------------------+------------------------+------------------------+------------------------+
| 76                     | 0.357                  | 0.973                  | nan                    | 7.011                  |
```

### PR DESCRIPTION
**Pull request by shantanuchhabra**
Fixes #1615

### CODE REVIEW

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 1]
```diff
@@ -148,26 +148,34 @@ def create(input_dataset, target, feature=None, validation_set='auto',
     classes = sorted(classes)
     class_to_index = {name: index for index, name in enumerate(classes)}
 
+    validation_set_corrective_string = ("'validation_set' parameter must be "
```
Comment (TobyRoseman) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: You should also be able to pass `None` if you want all of the data to be used for training.

Comment (TobyRoseman) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Look like you can do that. This error message makes it should like you can't.

Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Done.

File: src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 2]
```diff
@@ -656,7 +664,7 @@ def evaluate(self, dataset, metric = 'auto', verbose = True):
         
         return ret
 
-    def predict(self, data, verbose = True):
+    def predict(self, data, output_type='class', verbose=True):
```
Comment (nickjong) [inline src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: What does this have to do with improved error handing for validation sets? Please provide focuses PRs that only include the commits that need review. Otherwise, it's too easy for changes to get lost in the mix of what changes are duplicated and should be ignored. It invites sloppier and less accurate reviews.

Comment (shantanuchhabra) [reply src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Sorry about that, rogue commit made its way into this branch. Deleted it now.

**Comment by shantanuchhabra** [comment]
Internal tests pass.
